clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/StoreIngestionTask.java

```diff
@@ -1555,7 +1569,7 @@ protected void checkIngestionProgress(Store store) throws InterruptedException {
      * Check whether current consumer has any subscription or not since 'poll' function will throw
      * {@link IllegalStateException} with empty subscription.
      */
-    if (!consumerHasAnySubscription()) {
+    if (!consumerHasAnySubscription() && !hasAnyPendingSubscription()) {
```

Comment:
Looks like We have several places to use these two conditions together, how about we extract them out and put it into a simple helper function?

`consumerHasAnySubscription() || hasAnyPendingSubscription()`

Reply:
Updated in the latest diff.
However, I need to do another refactoring after talking with @majisourav99 .
We can't rely on consumer subscriptions and pending consumer actions to conclude if the SIT is idle or not, but instead, check PartitionConsumptionStatusMap and pending consumer actions. Context:
If unsubscriptionAfterBatchPushComplete is enabled, consumers would be unsubscribed from the VT, but no changes to PCS map on purpose, because we want to keep SIT thread running in this case to process actions like LEADER_TO_STANDBY transition or kill job action.

Reply:
Checked with both @lluwm and @majisourav99 offline. We can't unify the check condition so far.
1. The cleanup service must check the PCS map instead of consumer subscription states to make decision of cleaning SIT
2. SIT must check consumer subscription state to increase idle counter so that "skipAfterBatchPushUnsubEnabled" short-circuit flag can be set.
Currently there is a bug that skipAfterBatchPushUnsubEnabled will never be set back to False after rebalance happens. Potentially we can unify the check by moving skipAfterBatchPushUnsubEnabled flag update outside of the idle counter maintenance logic after fixing the bug.